src/TBufferStyle.cpp

```diff
@@ -0,0 +1,1262 @@
+                            // the colour space id first
+                        if (haveColorSpaceId) {
+                            if (paraIndex + 2 < total) {
+                                // We have the color space id
+                                madeElements << parameterStrings.at(paraIndex + 2);
+                            }
+                            if (paraIndex + 3 < total) {
+                                // We have the red component
+                                madeElements << parameterStrings.at(paraIndex + 3);
+                            }
+                            if (paraIndex + 4 < total) {
+                                // We have the green component
+                                madeElements << parameterStrings.at(paraIndex + 4);
+                            }
+                            if (paraIndex + 5 < total) {
+                                // We have the blue component
+                                madeElements << parameterStrings.at(paraIndex + 5);
+                            }
+                        } else {
+                            // Fake an empty colour space id
+                            madeElements << QString();
+                            if (paraIndex + 2 < total) {
+                                // We have the red component
+                                madeElements << parameterStrings.at(paraIndex + 2);
+                            }
+                            if (paraIndex + 3 < total) {
+                                // We have the green component
+                                madeElements << parameterStrings.at(paraIndex + 3);
+                            }
+                            if (paraIndex + 4 < total) {
+                                // We have the blue component
+                                madeElements << parameterStrings.at(paraIndex + 4);
+                            }
+                        }
+
+                        // Move the index to consume the used values
+                        decodeSGR48(madeElements, false);
+                        paraIndex += (haveColorSpaceId ? 5 : 4);
+                        break;
+                    case 1: // This uses no extra arguments and, as it means
+                            // transparent, is no use to us
+                        [[fallthrough]];
+                    default:
+                        break;
+                    }
+                }
+                // End of if (parameterElements.at(0) == QLatin1String("48"))
+            } else if (parameterElements.at(0) == QLatin1String("4")) {
+                // New way of controlling underline
+                bool isOk = false;
+                int value = parameterElements.at(1).toInt(&isOk);
+                if (!isOk) {
+                    // missing value
+                    qDebug().noquote().nospace() << "TBuffer::decodeSGR(\"" << sequence
```

Comment:
:information_source: You ought to update the `qDebug()` messages to refer to the new location (class) for all these relocated methods...